homeassistant/components/knx/storage/config_store.py

```diff
@@ -62,14 +69,24 @@ async def load_data(self) -> None:
                 len(self.data["entities"]),
             )
 
+    def add_platform(
+        self, platform: Platform, controller: PlatformControllerBase
+    ) -> None:
+        """Add platform controller."""
+        self._platform_controllers[platform] = controller
+
     async def create_entity(
         self, platform: Platform, data: dict[str, Any]
     ) -> str | None:
         """Create a new entity."""
-        if platform not in self.async_add_entity:
-            raise ConfigStoreException(f"Entity platform not ready: {platform}")
+        try:
+            platform_controller = self._platform_controllers[platform]
+        except KeyError as err:
+            raise ConfigStoreException(
+                f"Entity platform not ready: {platform}"
+            ) from err
```

Comment:
I don't really know how to write a test against this.
All UI supported platforms should create and add their platform_controllers on integration load.
The websocket call triggering `create_entity` will be voluptuous validated to not create an unsupported platform. So in theory this can never raise. If it did, it would just forward the error message in the UI.

Would you prefer to remove the try-except block in this case, or should it stay?

Reply:
How is the supported platform validated in the websocket? Does it check that the platform is setup? Cause that's what is required here. If the config entry is loaded we can assume that platforms are setup. If the websocket API checks that the config entry is loaded would be enough.

Reply:
I'd test this by setting up the config entry and then unloading it, and then calling the websocket API.

Reply:
To be clear, I don't think we need to catch the error here but we need to handle an unloaded config entry.